api/jenkins.ts

```diff
@@ -0,0 +1,108 @@
+import got from "../libs/got";
+import { createBadgenHandler, PathArgs } from "../libs/create-badgen-handler";
+import humanizeDuration from "humanize-duration";
+
+export default createBadgenHandler({
+  title: "Jenkins",
+  examples: {
+    "/jenkins/last-build/jenkins.mono-project.com/job/test-mono-mainline/":
+      "Last build status",
+    "/jenkins/fix-time/jenkins.mono-project.com/job/test-mono-mainline/":
+      "Time taken to fix a broken build",
+    "/jenkins/broken-build/jenkins.mono-project.com/job/test-mono-mainline/":
+      "# of broken builds",
+  },
+  handlers: {
+    "/jenkins/last-build/:hostname/:job*": lastJobStatusHandler,
+    "/jenkins/fix-time/:hostname/:job*": buildFixTimeHandler,
+    "/jenkins/broken-build/:hostname/:job*": brokenBuildsHandler,
+  },
+});
+
+const shortEnglishHumanizer = humanizeDuration.humanizer({
+  language: "shortEn",
+  units: ["d", "h"],
+  round: true,
+  languages: {
+    shortEn: {
+      y: () => "y",
+      mo: () => "mo",
+      w: () => "w",
+      d: () => "d",
+      h: () => "h",
+      m: () => "m",
+      s: () => "s",
+      ms: () => "ms",
+    },
+  },
+});
+
+const statusToColor = (status: string) => {
+  return status.toUpperCase() == "SUCCESS" ? "green" : "red";
+};
+
+const brokenBuildsToColor = (count: number) => {
+  return count < 10 ? "green" : count < 20 ? "orange" : "red";
+};
+const buildFixTimeToColor = (hours: number) => {
+  return hours < 2 ? "green" : hours < 6 ? "orange" : "red";
+};
+
+async function jenkinsLastBuild({ hostname, job }: PathArgs) {
+  const endpoint = `https://${hostname}/${job}/lastBuild/api/json?tree=result,timestamp,estimatedDuration`;
+  return await got(endpoint).json<any>();
+}
+
+async function jenkinsBuilds({ hostname, job }: PathArgs) {
+  const endpoint = `https://${hostname}/${job}/api/json?tree=builds[number,status,timestamp,id,result]`;
```

Comment:
Do we need to handle paging here? Would it return a quite large json if there's a lot builds?

Reply:
We can; however, the default behaviour of API restrict the response to 100 entries.